src/util/db/dbid.h

```diff
@@ -22,21 +22,27 @@
 // not add any additional state (= member variables). Inheritance is
 // only needed for type-safety.
 class DbId {
-  protected:
   public:
     constexpr DbId()
             : m_value(kInvalidValue) {
     }
+    explicit DbId(const int value)
+            : m_value(value) {
+        DEBUG_ASSERT(isValid() || (kInvalidValue == m_value));
```

Comment:
This assert looks weird and feels like it will always pass. As I understand, it should fail for `m_value < -1`. Maybe we can put that explicitly to help with readability?